app/controllers/api/v1/trips_controller.rb

```diff
@@ -0,0 +1,40 @@
+class Api::V1::TripsController < ApplicationController
+  before_action :set_trip, except: [:index]
+
+  def index
+    @trips = Trip.all
+    render json: @trips
+  end
+
+  def show
+    render json: @trip
+  end
+
+  def create
+    @trip = Trip.create(decoded_params)
+    render json: @trip
+  end
+
+  def update
+    @trip.update_attributes(decoded_params)
+    render json: @trip
+  end
+
+  def delete
+    @trip.destroy
+    render json: @trip
+  end
+
+
+  private
+
+  def set_trip
+    @trip = Trip.find(params[:id])
+  end
+
+  def decoded_params
+    ActiveSupport::JSON.decode(request.body.read)
```

Comment:
This is interesting, I haven't seen this syntax. Is the result similar to JSON.parse?